ArduPlane/quadplane.cpp

```diff
@@ -1937,7 +1937,38 @@ void QuadPlane::motors_output(bool run_rate_controller)
         if (now - last_att_control_ms > 100) {
             // relax if have been inactive
             relax_attitude_control();
+            /*
+              when starting the VTOL motors force use of the fixed
+              wing controller for attitude control if we are outside
+              of the Q_ANGLE_MAX limit. We keep it forced until we
+              have recovered to within the limit
+
+              This avoids an issue with the attitude controller
+              limiting the rate of recovery and also an issue with the
+              quaternion controller trying to fix the attitude in a
+              way that is bad for fixed wing aircraft (eg. putting
+              nose down when inverted, leading to rapid loss of
+              height)
+
+              we don't do this in QACRO or tailsitter modes, as this
+              type of attitude is expected
+            */
+            if (!tailsitter.enabled() &&
+                plane.control_mode != &plane.mode_qacro &&
+                (abs(ahrs_view->roll_sensor) > aparm.angle_max ||
+                 abs(ahrs_view->pitch_sensor) > aparm.angle_max)) {
+                force_fw_control_recovery = true;
+            }
         }
+        if (force_fw_control_recovery) {
+            // see if we have recovered attitude
+            if (abs(ahrs_view->roll_sensor) <= aparm.angle_max &&
+                abs(ahrs_view->pitch_sensor) <= aparm.angle_max) {
```

Comment:
When using the fixed wing controller it is flying to the normal AHRS (not the view). So using the view to check for completion may never work. For example if I had a `Q_TRIM_PITCH` of 15 and a `Q_ANGLE_MAX` of 10 this check would not pass.

Reply:
good catch, thanks, fixed